x-pack/plugins/security_solution/public/common/components/event_details/insights/insights.test.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+
+import { TestProviders } from '../../../mock';
+
+import { useKibana as mockUseKibana } from '../../../lib/kibana/__mocks__';
+import { useGetUserCasesPermissions } from '../../../lib/kibana';
+import { noCasesPermissions, readCasesPermissions } from '../../../../cases_test_utils';
+import { Insights } from './insights';
+import * as i18n from './translations';
+
+const mockedUseKibana = mockUseKibana();
+jest.mock('../../../lib/kibana', () => {
+  const original = jest.requireActual('../../../lib/kibana');
+
+  return {
+    ...original,
+    useGetUserCasesPermissions: jest.fn(),
+    useToasts: jest.fn().mockReturnValue({ addWarning: jest.fn() }),
+    useKibana: () => ({
+      ...mockedUseKibana,
+      services: {
+        ...mockedUseKibana.services,
+        cases: {
+          api: {
+            getRelatedCases: jest.fn(),
+          },
+        },
+      },
+    }),
+  };
+});
+const mockUseGetUserCasesPermissions = useGetUserCasesPermissions as jest.Mock;
+
+describe('Insights', () => {
+  beforeEach(() => {
+    mockUseGetUserCasesPermissions.mockReturnValue(noCasesPermissions());
+  });
+
+  it('does not render when there is no content to show', () => {
+    render(
+      <TestProviders>
+        <Insights browserFields={{}} eventId="test" data={[]} timelineId="" />
+      </TestProviders>
+    );
+
+    expect(
+      screen.queryByRole('heading', {
+        name: i18n.INSIGHTS,
+      })
+    ).not.toBeInTheDocument();
+  });
+
+  it('renders when there is at least one insight element to show', () => {
+    mockUseGetUserCasesPermissions.mockReturnValue(readCasesPermissions());
```

Comment:
It's not quickly evident from the way the test is written that there is "something to show". Maybe adding a comment stating cases permissions allows us to display the cases section whether or not there are actually cases to show

Reply:
Does this explain it? https://github.com/elastic/kibana/pull/136009/commits/0c3cdcdd99df85fedb7aa71b73019a03b536a9c0

:)